local_groups/models.py

```diff
@@ -56,65 +145,132 @@ class Group(models.Model):
 
     state_organizing_committee = models.BooleanField(default=False)
 
     STATUSES = (
-       ('submitted', 'Submitted'),
-       ('signed-mou', 'Signed MOU'),
-       ('inactive', 'Inactive'),
-       ('approved', 'Approved'),
-       ('removed', 'Removed') # can flesh out later
+        ('submitted', 'Submitted'),
+        ('signed-mou', 'Signed MOU'),
+        ('inactive', 'Inactive'),
+        ('approved', 'Approved'),
+        ('removed', 'Removed')
+    )
+    status = models.CharField(
+        max_length=64,
+        choices=STATUSES,
+        default='submitted'
     )
-    status = models.CharField(max_length=64, choices=STATUSES, default='submitted')
 
     VERSIONS = (
-       ('none', 'N/A'),
-       ('1.0', 'Old'),
-       ('1.1', 'Current'),
-   )
+        ('none', 'N/A'),
+        ('1.0', 'Old'),
+        ('1.1', 'Current'),
+    )
 
-    signed_mou_version = models.CharField(max_length=64, choices=VERSIONS, default='none', verbose_name='MOU Version', null=True, blank=True)
+    signed_mou_version = models.CharField(
+        max_length=64,
+        choices=VERSIONS,
+        default='none',
+        verbose_name='MOU Version',
+        null=True,
+        blank=True
+    )
 
     ORGANIZERS = (
-       ('juliana', 'Juliana'),
-       ('basi', 'Basi'),
-       ('kyle', 'Kyle'),
+        ('juliana', 'Juliana'),
+        ('basi', 'Basi'),
+        ('kyle', 'Kyle'),
     )
 
-    organizer = models.CharField(max_length=64, choices=ORGANIZERS, default=None, verbose_name='Organizer', null=True, blank=True)
+    organizer = models.CharField(
+        max_length=64,
+        choices=ORGANIZERS,
+        default=None,
+        verbose_name='Organizer',
+        null=True,
+        blank=True
+    )
 
-    mou_url = models.URLField(null=True, blank=True, verbose_name="MOU URL", max_length=255)
+    mou_url = models.URLField(
+        null=True,
+        blank=True,
+        verbose_name="MOU URL",
+        max_length=255
+    )
 
     # Notes field for internal OR staff use
     notes = models.TextField(
         blank=True,
-        help_text=  """Please include dates here along with notes to make
-                    reporting easier.""",
+        help_text="""Please include dates here along with notes to make
+            reporting easier.""",
         null=True,
         verbose_name="Notes"
     )
 
+    def save(self, *args, **kwargs):
+        # TODO: make main groups url an environment variable
+        # and replace hardcoded /groups throughout site
 
-    # source = models.CharField(max_length=17, null=True, blank=True)
-    # subsource = models.FloatField(null=True, blank=True)
-    # i_p__address = models.CharField(max_length=16, null=True, blank=True)
-    # constituent__id = models.IntegerField(null=True, blank=True)
-    # signup__id = models.IntegerField(null=True, blank=True)
+        purge_url_from_cache('/groups')
+        purge_url_from_cache('/groups/' + self.slug)
+        super(Group, self).save(*args, **kwargs)
```

Comment:
according to docs should be able to clear cache after save:
```

    def save(self, *args, **kwargs):
        do_something()
        super(Blog, self).save(*args, **kwargs) # Call the "real" save() method.
        do_something_else()
```
https://docs.djangoproject.com/en/1.10/topics/db/models/#overriding-predefined-model-methods